NightscoutServiceKit/NightscoutService.swift

```diff
@@ -59,432 +104,215 @@ public final class NightscoutService: Service {
-                return "SingleUp"
-            case .upUp?, .upUpUp?:
-                return "DoubleUp"
-            case .down?:
-                return "SingleDown"
-            case .downDown?, .downDownDown?:
-                return "DoubleDown"
-            case .flat?:
-                return "Flat"
-            case .none:
-                return nil
+        // TODO: Prevent reentrancy
+        synchronizeStatusRemoteData { result in
+            switch result {
+            case .failure(let error):
+                completion(.failure(error))
+            case .success(let uploadedStatus):
+                self.synchronizeSettingsRemoteData { result in
+                    switch result {
+                    case .failure(let error):
+                        completion(.failure(error))
+                    case .success(let uploadSettings):
+                        self.synchronizeGlucoseRemoteData { result in
+                            switch result {
+                            case .failure(let error):
+                                completion(.failure(error))
+                            case .success(let uploadedGlucose):
+                                self.synchronizeDoseRemoteData { result in
+                                    switch result {
+                                    case .failure(let error):
+                                        completion(.failure(error))
+                                    case .success(let uploadedDose):
+                                        self.synchronizeCarbRemoteData { result in
+                                            switch result {
+                                            case .failure(let error):
+                                                completion(.failure(error))
+                                            case .success(let uploadedCarb):
+                                                completion(.success(uploadedStatus || uploadSettings || uploadedGlucose || uploadedDose || uploadedCarb))
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
             }
-        }()
-
-        for value in values {
-            uploader.uploadSGV(
-                glucoseMGDL: Int(value.quantity.doubleValue(for: .milligramsPerDeciliter)),
-                at: value.startDate,
-                direction: direction,
-                device: device
-            )
         }
     }
 
-    public func upload(pumpEvents events: [PersistedPumpEvent], fromSource source: String, completion: @escaping (Result<[URL], Error>) -> Void) {
-        guard let uploader = uploader else {
-            completion(.success(events.map({ $0.objectIDURL })))
-            return
+    private func synchronizeStatusRemoteData(completion: @escaping (Result<Bool, Error>) -> Void) {
```

Comment:
This method (and its associated persisted query anchors) is a candidate for refactoring up into the Loop layer, I think. Maybe we can talk this through. Doing so would reduce the responsibilities of the service in a significant way, decoupling them from any kind of interaction with a data store.
It's also functionality that is likely to be copied across each remote data service.

Reply:
Done.